airflow/www/alias-rest-types.js

```diff
@@ -0,0 +1,185 @@
+// This is recurrsivenly called to generate types for Linege's external dependencies.
+const generateAliases = (rootNode, writeText, prefix = '') => {
+  // Loop through the root AST nodes of the file
+  ts.forEachChild(rootNode, (node) => {
+    // Response Data Types
+    if (ts.isInterfaceDeclaration(node) && node.name?.text === 'components') {
+      const schemaMemberNames = findNode(node, 'schemas').type.members.map((n) => n.name?.text);
+
+      const types = schemaMemberNames.map((n) => [
+        `${n}`,
+        `export type ${n} = ${prefixPath(prefix, 'components')}['schemas']['${n}'];`,
+      ]);
+      if (types.length) {
+        writeText.push(['comment', `Types for returned data ${prefix}`]);
+        writeText.push(...types);
+      }
+    }
+
+    /* Fetch Query Variable Types
+     * These are the variables that are passed to the fetch function, comprising variables that
+     * may go into the path (`/namespaces/{namespace}/datasets/{dataset}/metrics`)
+     * and those that are passed in as query parameters.
+     */
+
+    // Paths referencing an operation are skipped
+    if (node.name?.text === 'paths') {
+      if (!prefix) {
+        writeText.push(['comment', 'Alias paths to PascalCase.']);
+        writeText.push(['Paths', 'export type Paths = paths;']);
+      }
+
+      const types = [];
+
+      (node.members || node.type.members).forEach((path) => {
+        const methodNames = path.type.members.map((m) => m.name.text);
+        const methodTypes = methodNames.map((m) => (
+          generateVariableAliases(
+            findNode(path, m),
+            `${prefixPath(prefix, 'paths')}['${path.name?.text}']['${m}']`,
+            `${path.name.text}${toPascalCase(m)}`,
+          )));
+        types.push(...methodTypes.filter((m) => !!m));
+      });
+
+      if (types.length) {
+        writeText.push(['comment', `Types for path operation variables ${prefix}`]);
+        writeText.push(...types);
+      }
+    }
+
+    // operationIds are defined
+    if (node.name?.text === 'operations') {
+      if (!prefix) {
+        writeText.push(['comment', 'Alias operations to PascalCase.']);
+        writeText.push(['Operations', 'export type Operations = operations;']);
+      }
+
+      const types = (node.members || node.type.members).map((operation) => (
+        generateVariableAliases(
+          operation,
+          `${prefixPath(prefix, 'operations')}['${operation.name.text}']`,
+          operation.name.text,
+        )));
+      if (types.length) {
+        writeText.push(['comment', `Types for operation variables ${prefix}`]);
+        writeText.push(...types);
+      }
+    }
+
+    // recursively call this for any externals
+    if (ts.isInterfaceDeclaration(node) && node.name?.text === 'external') {
+      node.members.forEach((external) => {
+        generateAliases(external.type, writeText, `external['${external.name.text}']`);
+      });
+    }
+  });
+};
+
+function generate(file) {
```

Comment:
Should we also push the licence directly, or are we counting on the pre-commit hook to do that for us ?